c/tskit/core.h

```diff
@@ -189,17 +207,25 @@ to the API or ABI are introduced, i.e., internal refactors of bugfixes.
  * options for individual functions at the bottom without worrying about
  * clashing with the common options */
 
-/** @brief Turn on debugging output. Not supported by all functions. */
+/** Turn on debugging output. Not supported by all functions. */
 #define TSK_DEBUG (1u << 31)
 
-/** @brief Do not initialise the parameter object. */
+/** Do not initialise the parameter object. */
 #define TSK_NO_INIT (1u << 30)
 
-/** @brief Do not run integrity checks before performing an operation. */
+/**
+Do not run integrity checks before performing an operation.
+This performance optimisation should not be used unless the calling code can
+guarantee reference integrity within the table collection. References
+to rows not in the table or bad offsets will result in undefined
+behaviour.
+*/
 #define TSK_NO_CHECK_INTEGRITY (1u << 29)
 
-/** @brief Instead of taking a copy of input data, take ownership of it
- *  and its lifecycle */
+/** Instead of taking a copy of input objects, the function should take ownership
```

Comment:
Some inconsistency here in the comment style. Probably better to avoid the javadoc style start-each-line-with-* form